app/models/mhv_account.rb

```diff
@@ -6,71 +6,74 @@ class MhvAccount < ActiveRecord::Base
   include AASM
 
   TERMS_AND_CONDITIONS_NAME = 'mhvac'
-  # Everything except existing and ineligible accounts should be able to transition to :needs_terms_acceptance
-  ALL_STATES = %i[
-    unknown
-    needs_terms_acceptance
-    existing
-    ineligible
-    registered
-    upgraded
-    register_failed
-    upgrade_failed
+  INELIGIBLE_STATES = %i[
+    needs_identity_verification needs_ssn_resolution needs_va_patient
+    has_deactivated_mhv_ids has_multiple_active_mhv_ids
+    state_ineligible country_ineligible needs_terms_acceptance
   ].freeze
 
+  FAILURE_STATES = %i[register_failed upgrade_failed].freeze
+  PERSISTED_STATES = %i[registered upgraded].freeze
+  ELIGIBLE_STATES = %i[existing eligible no_account].freeze
+  ALL_STATES = (%i[unknown] + FAILURE_STATES + INELIGIBLE_STATES + ELIGIBLE_STATES + PERSISTED_STATES).freeze
+
   after_initialize :setup
 
+  # rubocop:disable Metrics/BlockLength
   aasm(:account_state) do
     state :unknown, initial: true
-    state :needs_terms_acceptance, :existing, :ineligible, :registered, :upgraded, :register_failed, :upgrade_failed
+    state(*ALL_STATES)
 
+    # NOTE: This is eligibility for account creation or upgrade, not for access to services.
     event :check_eligibility do
-      transitions from: ALL_STATES, to: :existing, if: :preexisting_account?
-      transitions from: ALL_STATES, to: :ineligible, unless: :eligible?
-      transitions from: ALL_STATES, to: :upgraded, if: :previously_upgraded?
-      transitions from: ALL_STATES, to: :registered, if: :previously_registered?
-      transitions from: ALL_STATES, to: :unknown
+      transitions from: ALL_STATES, to: :needs_identity_verification, unless: :identity_proofed?
+      transitions from: ALL_STATES, to: :needs_ssn_resolution, if: :ssn_mismatch?
+      transitions from: ALL_STATES, to: :needs_va_patient, unless: :va_patient?
+      transitions from: ALL_STATES, to: :has_deactivated_mhv_ids, if: :deactivated_mhv_ids?
+      transitions from: ALL_STATES, to: :has_multiple_active_mhv_ids, if: :multiple_active_mhv_ids?
+      transitions from: ALL_STATES, to: :needs_terms_acceptance, unless: :terms_and_conditions_accepted?
```

Comment:
Apparently we don't want to ask for TC acceptance if the user has an existing Premium account (
department-of-veterans-affairs/vets.gov-team#10533), so we should update the conditional here.